libs/wingsdk/src/cloud/function.ts

```diff
@@ -115,10 +115,13 @@ export class Function extends Resource implements IInflightHost {
   protected _getCodeLines(handler: IFunctionHandler): string[] {
     const inflightClient = handler._toInflight();
     const lines = new Array<string>();
+    const client = "$handler";
 
     lines.push('"use strict";');
+    lines.push(`var ${client} = undefined;`);
     lines.push("exports.handler = async function(event) {");
-    lines.push(`  return await (${inflightClient}).handle(event);`);
+    lines.push(`  ${client} = ${client} ?? (${inflightClient});`);
```

Comment:
This means that the the `inflight new` will only be executed once per cold start. Since cold starts aren't strongly defined, we can say the the inflight instance will be created and execute its `inflight new` at least once per `cloud.Function` instance, but possibly multiple times and always before your handler is executed. We might want to document this.

Reply:
yes, that's the behavior in AWS Lambda...

I'll add to the docs.